Framework/Kernel/src/DiskBuffer.cpp

```diff
@@ -52,19 +52,18 @@ void DiskBuffer::toWrite(ISaveable *item) {
                              // its size in memory
   {
     // forget old memory size
-    m_mutex.lock();
+    std::unique_lock<std::mutex> uniqueLock(m_mutex);
```

Comment:
Would you be better off with a `std::lock_guard` that goes out of scope one line later at the `}`?